### PR TITLE
Add FIDO biometric authentication as a use case

### PR DESCRIPTION
**Pull request by laurencelundblade**
Address #135.

There are on the order of 1 billion users of biometric authentication

### CODE REVIEW

File: draft-ietf-rats-architecture.md [hunk 1]
```diff
@@ -316,6 +316,26 @@ Relying Party:
 : A remote server that will securely grant the Attester
 permission to continue operating (i.e., not reboot) for a period of time
 
+## FIDO Biometric Authentication
+
+In the FIDO protocol, it is the device in the user’s hand that authenticates the human user by biometrics like fingerprints or PIN and password.
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:321]: (simplify grammar)
```suggestion
In the Fast IDentity Online (FIDO) protocol, the device in the user’s hand authenticates the human user by biometrics such as fingerprints or PIN and password.
```

Also can we add a citation for the protocol?

File: draft-ietf-rats-architecture.md [hunk 2]
```diff
@@ -316,6 +316,26 @@ Relying Party:
...
+The comparison of the input from the user to the reference or expected values is done by the device, often a mobile phone, not the relying party.
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:322]: So the device is a verifier?  (not sure why "not the relying party" is in the sentence)

File: draft-ietf-rats-architecture.md [hunk 3]
```diff
@@ -316,6 +316,26 @@ Relying Party:
...
+For the relying party to know that the authentication is trustworthy the relying party needs to know that the authenticator part of the device is trustworthy.
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:323]:
```suggestion
For the relying party to know that the authentication is trustworthy, the relying party needs to know that the authenticator part of the device is trustworthy.
```

File: draft-ietf-rats-architecture.md [hunk 4]
```diff
@@ -316,6 +316,26 @@ Relying Party:
...
+The FIDO protocol employs attestation for this. 
+
+The FIDO protocol supports several attestation protocols and a mechanism by which new ones can be registered and added. Attestation defined by RATS is thus a candidate for use by FIDO.
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:326]: meaning, for use by the FIDO protocol?  or for use by the FIDO Alliance?

File: draft-ietf-rats-architecture.md [hunk 5]
```diff
@@ -316,6 +316,26 @@ Relying Party:
...
+## FIDO Biometric Authentication
+
+In the FIDO protocol, it is the device in the user’s hand that authenticates the human user by biometrics like fingerprints or PIN and password.
+The comparison of the input from the user to the reference or expected values is done by the device, often a mobile phone, not the relying party.
+For the relying party to know that the authentication is trustworthy the relying party needs to know that the authenticator part of the device is trustworthy.
+The FIDO protocol employs attestation for this. 
+
+The FIDO protocol supports several attestation protocols and a mechanism by which new ones can be registered and added. Attestation defined by RATS is thus a candidate for use by FIDO. 
+
+Other biometric authentication protocols such as the Chinese IFAA standard and WeChat Pay as well as Google Pay make use of attestation in one form or another.
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:328]: Can we add citations to these protocols if we keep this sentence?

File: draft-ietf-rats-architecture.md [hunk 6]
```diff
@@ -316,6 +316,26 @@ Relying Party:
...
+
+Attester: 
+
+:The FIDO Authenticator.
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:332]: "authenticator" was lower case in 323. Pick one.

File: draft-ietf-rats-architecture.md (hunk 1 above)
Comment (henkbirkholz) [reply draft-ietf-rats-architecture.md:321]: Fine with the change. I think that this is *the* most important value add of FIDO: the Authenticator component that can require a human to be in the loop.

File: draft-ietf-rats-architecture.md (hunk 2 above)
Comment (henkbirkholz) [reply draft-ietf-rats-architecture.md:322]: The Authenticator is a component of the Attester, isn't it? It is used (in the weakest sense) for evidence generation, I think.

File: draft-ietf-rats-architecture.md [hunk 7]
```diff
@@ -316,6 +316,26 @@ Relying Party:
 : A remote server that will securely grant the Attester
 permission to continue operating (i.e., not reboot) for a period of time
 
+## FIDO Biometric Authentication
+
+In the FIDO protocol, it is the device in the user’s hand that authenticates the human user by biometrics like fingerprints or PIN and password.
+The comparison of the input from the user to the reference or expected values is done by the device, often a mobile phone, not the relying party.
+For the relying party to know that the authentication is trustworthy the relying party needs to know that the authenticator part of the device is trustworthy.
+The FIDO protocol employs attestation for this.
```
Comment (henkbirkholz) [inline draft-ietf-rats-architecture.md:324]:
```suggestion
The FIDO protocol employs evidence generation for this. 
```

Comment (henkbirkholz) [reply draft-ietf-rats-architecture.md:324]: I know... that is already standard lingo, but I really think that Attestation is too confusing and does not add meaning here.

File: draft-ietf-rats-architecture.md (hunk 6 above)
Comment (henkbirkholz) [reply draft-ietf-rats-architecture.md:332]: I do not think hat the Authenticator component alone is the Attester here. It provides strong Authentication to be used in Evidence generated by the device though.

File: draft-ietf-rats-architecture.md (hunk 7 above)
Comment (laurencelundblade) [reply draft-ietf-rats-architecture.md:324]: The FIDO protocol does employ attestation. The FIDO protocol defines an attestation protocol of its own. It also can use other attestation protocols.

Conceptually, a FIDO authenticator first runs an attestation protocol to prove itself trustworthy to the relying party (no separation of Verifier and RP in FIDO). Second, it authenticates the user and tells the RP the result. When it is telling the RP the result, it is not running an attestation protocol.

File: draft-ietf-rats-architecture.md (hunk 4 above)
Comment (laurencelundblade) [reply draft-ietf-rats-architecture.md]: protocol; addressed in recent commit.

File: draft-ietf-rats-architecture.md (hunk 2 above)
Comment (laurencelundblade) [reply draft-ietf-rats-architecture.md]: The device verifies biometric input locally. Nothing to do with verifying attestation evidence. Hopefully recent commit addresses this.

File: draft-ietf-rats-architecture.md [hunk 8]
```diff
@@ -78,6 +78,14 @@ informative:
     author:
       org: "Trusted Computing Group"
     target: "https://trustedcomputinggroup.org/wp-content/uploads/TCG_TPM2_r1p62_Part1_Architecture_7july2020.pdf"
+  WebAuthN:
+    title: Web Authentication: An API for accessing Public Key Credentials
+    author: The W3C
+    target: https://www.w3.org/TR/webauthn-1/ 
+  CTAP:
+    title: Client to Authenticator Protocol
+    author: The FIDO Alliance
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:87]:
```suggestion
    author: FIDO Alliance
```
(The copyright at the bottom of the linked URL just says "FIDO Alliance", no "The" in front)

File: draft-ietf-rats-architecture.md [hunk 9]
```diff
@@ -78,6 +78,14 @@ informative:
     author:
       org: "Trusted Computing Group"
     target: "https://trustedcomputinggroup.org/wp-content/uploads/TCG_TPM2_r1p62_Part1_Architecture_7july2020.pdf"
+  WebAuthN:
+    title: Web Authentication: An API for accessing Public Key Credentials
+    author: The W3C
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:83]:
```suggestion
    author: W3C
```
(copyright statement in linked document just says "W3C", with no "The")

File: draft-ietf-rats-architecture.md [hunk 10]
```diff
@@ -316,6 +324,26 @@ Relying Party:
 : A remote server that will securely grant the Attester
 permission to continue operating (i.e., not reboot) for a period of time
 
+## FIDO Biometric Authentication
+
+In the Fast IDentity Online (FIDO) protocol {{WebAuthN}}, {{CTAP}}, the device in the user’s hand authenticates the human user, whether by biometrics (such as fingerprints), or by PIN and password.
+FIDO authentication puts a large amount of trust in the device compared to typical password authentication because it is the device that verifies the biometric, PIN and password inputs from the user, not the server.
+For the relying party to know that the authentication is trustworthy, the relying party needs to know that the Authenticator part of the device is trustworthy.
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:331]:
```suggestion
For the Relying Party to know that the authentication is trustworthy, the Relying Party needs to know that the Authenticator part of the device is trustworthy.
```

File: draft-ietf-rats-architecture.md (hunk 5 above)
Comment (laurencelundblade) [reply draft-ietf-rats-architecture.md:336]: I added references for FIDO.

There aren't protocol documents in English for IFAA and WeChat Pay, but there are fluffy marketing overviews. So far citations for the other use cases haven't been required, so this seems OK.

File: draft-ietf-rats-architecture.md (hunk 7 above)
Comment (dthaler) [reply draft-ietf-rats-architecture.md:332]: Laurence's explanation makes sense to me, I am fine with "remote attestation" in this sentence (but add "remote").

File: draft-ietf-rats-architecture.md [hunk 11]
```diff
@@ -316,6 +324,26 @@ Relying Party:
...
+The FIDO protocol employs attestation for this. 
+
+The FIDO protocol supports several attestation protocols and a mechanism by which new ones can be registered and added. Attestation defined by RATS is thus a candidate for use in the FIDO protocol.
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:334]:
```suggestion
The FIDO protocol supports several remote attestation protocols and a mechanism by which new ones can be registered and added. Remote attestation defined by RATS is thus a candidate for use in the FIDO protocol. 
```

File: draft-ietf-rats-architecture.md (hunk 5 above)
Comment (dthaler) [reply draft-ietf-rats-architecture.md:336]: Agreed, thanks for checking.

**Comment by dthaler** [comment]
Discussion: everything you can do, RATS can do better :)  Aka every authentication/access control use case is an attestation use case.  Laurence to take a first shot at adding text into intro above use cases.

Details would be in a doc from FIDO or W3C.

**Pull request merged**